exemption.php

```diff
@@ -0,0 +1,73 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Page to manage user exemptions.
+ *
+ * @package     factor_exemption
+ * @author      Peter Burnett <peterburnett@catalyst-au.net>
+ * @copyright   Catalyst IT 2023
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+require_once(__DIR__ . '/../../../../../config.php'); require_once($CFG->libdir . '/adminlib.php');
+
+$delete = optional_param('delete', 0, PARAM_INT);
+$extend = optional_param('extend', 0, PARAM_INT);
+
+$context = \context_system::instance();
+$PAGE->set_context($context);
+// We need to load the full admin tree into memory here, otherwise our node isnt locatable as it is loaded conditionally.
+admin_get_root(false, true);
+admin_externalpage_setup('factorexemptionmanageexemptions');
```

Comment:
Just double checking, this is what does the permissions check for this page ?

Reply:
correct, it enforces moodle/site:config and does require_login()